### PR TITLE
[2201.2.x] Improve obj field mapping constructor completions 

### LINKED ISSUE #37625: [Bug]: Missing completions for record fields when used in object fields
### Description

1. Missing completions for `x`, `y` and `Fill Required Fields`.
Screenshot 2022-09-05 at 12 15 06

Expected completions :
Screenshot 2022-09-05 at 12 16 56

2. Missing completions for fields of `Employee`.
Screenshot 2022-09-05 at 12 23 05

Cause seems to be that the resolved type is empty in https://github.com/ballerina-platform/ballerina-lang/blob/fd504a0e45d98daec0eaeaa334e9793f53edc41c/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MappingContextProvider.java#L107

### Steps to Reproduce

1.

```
class testClass {
    record {int y; int x;} rec = {};
}
```

2.

```
type Employee record {
    readonly string name;
    int salary;
};

class testClass {
    table<Employee> key(name) emp = table [{}];
}
```

### Affected Version(s)

2201.2.0 (Swan Lake Update 2)

### OS, DB, other environment details and versions

_No response_

### Labels

LanguageServer

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Fixes #37625

Related PRs - #38367 (2201.3.x), #38370 (master)

## Samples
Screenshot 2022-09-15 at 09 49 16

Screenshot 2022-09-15 at 09 49 51

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/resources/completion/expression_context/config/mapping_constructor_expr_ctx_config1.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+  "position": {
+    "line": 1,
+    "character": 34
+  },
+  "source": "expression_context/source/mapping_constructor_expr_ctx_source1.bal",
```
Comment (gayalkuruppu) [inline language-server/modules/langserver-core/src/test/resources/completion/expression_context/config/mapping_constructor_expr_ctx_config1.json:6]: Shall we add a description here?

**Review by gayalkuruppu — COMMENTED** [review]
Changes look good to me. Just a small suggestion to add the description field to the config files.

**Review by gayalkuruppu — DISMISSED** [review]
LGTM

**Review by malinthar — CHANGES_REQUESTED** [review]
Shall we address the following cases as well. 

1. Suggest variable names(object field names) in mapping constructor
```Ballerina
int moduleVar = 20;

class MyClass {
    int x = 10;
    int y = 20;
    record {int y; int x;} rec = {x, <cursor>};
}
```
Here we should get a completion item for `y`(object fields) as we have done other variables.
image




```
2. Provide object fields for the value expression
![image](https://user-images.githubusercontent.com/35211477/191147900-870b450b-1d50-4cf5-a2a3-612eca0a2b43.png)

we should include `x` and `y` in the completion item list

**Comment by LakshanWeerasinghe** [comment]
> Shall we address the following cases as well.
> 
> 1. Suggest variable names(object field names) in mapping constructor
> 
> ```ballerina
> int moduleVar = 20;
> 
> class MyClass {
>     int x = 10;
>     int y = 20;
>     record {int y; int x;} rec = {x, <cursor>};
> }
> ```
> 
> Here we should get a completion item for `y`(object fields) as we have done other variables. image
> 
> 2. Provide object fields for the value expression
>    image
> 
> we should include `x` and `y` in the completion item list

I have add these two scenarios.

**Comment by LakshanWeerasinghe** [comment]
Closing this PR in favor of #38367 (2201.3.x) and #38370 (master).

**Pull request closed without merging**